Assets/Stack mini game/Scripts/GameState.cs

```diff
@@ -8,17 +8,26 @@
 [Serializable]
 public class GameState
 {
-    [SerializeField] private StackMiniGame_StateRegister stateRegister;
+    [SerializeField] private StateRegister stateRegister;
 
-    public string CurrentState { get; private set; }
+    private string currentState;
 
     public void SetCurrentState(string newState)
     {
         if(TryGetState(newState.ToLower().Trim(), out string result) == true)
-                CurrentState = result;
+                currentState = result;
     }
 
-    public bool TryGetState(string state, out string result)
+    public bool CheckCurrentState(string state)
+    {
+        string result;
+
+        TryGetState(state, out result);
+
+        return currentState == result;
+    }
+

```

Comment:
Since you do in the other classes 

```
if (gameState.CheckCurrentState("full win") || 
                 gameState.CheckCurrentState("partial win") || 
                 gameState.CheckCurrentState("lose"))
```

Maybe here you could have in here:

```csharp
    public bool CheckCurrentState(IEnumerable<string> states)
    {
        // Do any of the states passed in match the current state?
        return states.Any(state => TryGetState(state, out string result) && currentState == result);
    }
```